src/core/layout/qgscompositionconverter.h

```diff
@@ -0,0 +1,231 @@
+class QgsLayoutItemMap;
+class QgsLayoutItemScaleBar;
+class QgsLayoutItemLegend;
+class QgsLayoutItemHtml;
+class QgsLayoutItemAttributeTable;
+
+
+/**
+ * QgsCompositionConverter class converts a QGIS 2.x composition to a QGIS 3.x layout
+ * \since QGIS 3.0
+ * \note Not available in Python bindings.
+ * \ingroup core
+ */
+class CORE_EXPORT QgsCompositionConverter
+{
+  public:
+
+    /**
+     * Composition data defined properties for different item types
+     */
+    enum DataDefinedProperty
+    {
+      NoProperty = 0, //!< No property
+      AllProperties, //!< All properties for item
+      TestProperty, //!< Dummy property with no effect on item
+      //composer page properties
+      PresetPaperSize, //!< Preset paper size for composition
+      PaperWidth, //!< Paper width
+      PaperHeight, //!< Paper height
+      NumPages, //!< Number of pages in composition
+      PaperOrientation, //!< Paper orientation
+      //general composer item properties
+      PageNumber, //!< Page number for item placement
+      PositionX, //!< X position on page
+      PositionY, //!< Y position on page
+      ItemWidth, //!< Width of item
+      ItemHeight, //!< Height of item
+      ItemRotation, //!< Rotation of item
+      Transparency, //!< Item transparency (deprecated)
+      Opacity, //!< Item opacity
+      BlendMode, //!< Item blend mode
+      ExcludeFromExports, //!< Exclude item from exports
+      FrameColor, //!< Item frame color
+      BackgroundColor, //!< Item background color
+      //composer map
+      MapRotation, //!< Map rotation
+      MapScale, //!< Map scale
+      MapXMin, //!< Map extent x minimum
+      MapYMin, //!< Map extent y minimum
+      MapXMax, //!< Map extent x maximum
+      MapYMax, //!< Map extent y maximum
+      MapAtlasMargin, //!< Map atlas margin
+      MapLayers, //!< Map layer set
+      MapStylePreset, //!< Layer and style map theme
+      //composer picture
+      PictureSource, //!< Picture source url
+      PictureSvgBackgroundColor, //!< SVG background color
+      PictureSvgStrokeColor, //!< SVG stroke color
+      PictureSvgStrokeWidth, //!< SVG stroke width
+      //html item
+      SourceUrl, //!< Html source url
+      //legend item
+      LegendTitle, //!< Legend title
+      LegendColumnCount, //!< Legend column count
+      //scalebar item
+      ScalebarFillColor, //!< Scalebar fill color
+      ScalebarFillColor2, //!< Scalebar secondary fill color
+      ScalebarLineColor, //!< Scalebar line color
+      ScalebarLineWidth, //!< Scalebar line width
+    };
+
+    /**
+     * The MarkerMode enum is the old QGIS 2.x arrow marker mode
+     */
+    enum MarkerMode
+    {
+      DefaultMarker,
+      NoMarker,
+      SVGMarker
+    };
+
+
+    /**
+     * createLayoutFromCompositionXml is a factory that creates layout instances from a
+     *  QGIS 2.x XML composition \a parentElement and a QGIS \a project
+     * \return a QgsPrintLayout instance
+     * \since QGIS 3.0
```

Comment:
I think convention is to leave out the since annotations when it matches the class' version added.